Lib/argparse.py

```diff
@@ -325,11 +325,19 @@ def _format_usage(self, usage, actions, groups, prefix):
             if len(prefix) + len(usage) > text_width:
 
                 # break usage into wrappable parts
-                part_regexp = r'\(.*?\)+|\[.*?\]+|\S+'
+                part_regexp = '|'.join([
+                    r'\(.*?\)+\s',
+                    r'\[.*?\]+\s',
+                    r'\(.*?\)+$',
+                    r'\[.*?\]+$',
+                    r'\S+',
+                ])
```

Comment:
Minor: you can use automatic string concatenation (`'ab'  'c' == 'abc'), or a multiline regex with the re.M flag.

Reply:
I think the `'|'.join([...])` is OK because it shows the parts of the regex like this OR that OR the_other

Reply:
I believe Ezio's point was that you can get that benefit without the runtime cost of the join by writing it as:

    part_regexp = (
                   r'\(.*?\)+\s|'
                   r'\[.*?\]+\s|'
                   r'\(.*?\)+$|'
                   r'\[.*?\]+$|'
                   r'\S+'
               )